saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh

```diff
@@ -158,6 +167,14 @@ main() {
         start_nginx
       fi
     fi
+
+    INSTANCE_ID=
+    if [[ "$CLOUD_PLATFORM" == "AWS" ]]; then
+      INSTANCE_ID="`wget -q -O - http://169.254.169.254/latest/meta-data/instance-id`"
+    elif [[ "$CLOUD_PLATFORM" == "AZURE" ]]; then
+      INSTANCE_ID="`wget --header="Metadata: true" http://169.254.169.254/metadata/instance/compute/vmId?api-version=2017-08-01&format=text`"
```

Comment:
Does the quoting work correctly here? It might be better to use `$()` instead of backticks.

Also, does this command require the options `-q -O -` ?

Reply:
Double quotes inside backticks work as desired. I haven't been able to test the Azure size, but agree that those options seem desirable.

Reply:
wget gets installed as part of the prerequisites, for all providers except YCloud, so that should not be an issue